internal/linker/linker.go

```diff
@@ -36,9 +37,11 @@ var (
 	linkerPatchesFS embed.FS
 )
 
-func loadLinkerPatches() (string, map[string]string, error) {
+func loadLinkerPatches() (string, []string, []string, error) {
```

Comment:
use named result parameters, otherwise it's hard to tell what each of the three values `string, []string, []string` actually mean.

for example:

`(version string, modified, patches []string, error)`